pkg/kube/inject/webhook.go

```diff
@@ -608,10 +609,48 @@ func applyPrometheusMerge(pod *corev1.Pod, mesh *meshconfig.MeshConfig) error {
 		pod.Annotations["prometheus.io/port"] = targetPort
 		pod.Annotations["prometheus.io/path"] = "/stats/prometheus"
 		pod.Annotations["prometheus.io/scrape"] = "true"
+		return nil
 	}
+
 	return nil
 }
 
+var (
+	prometheusScrapeAnnotations = []string{
+		"prometheus.io/scrape",
+		"prometheus.io.scrape",
+	}
+
+	prometheusPortAnnotations = []string{
+		"prometheus.io/port",
+		"prometheus.io.port",
```

Comment:
Prometheus has a variety of charecters that it escapes to the same thing. I don't know the fully list, but ideally we normalize the annotations in the same way they do.

For example `prometheus/io/port` would work with prometheus but this logic would not handle it

Reply:
you're right, `prometheus/io/port` also will convert to `__meta_kubernetes_pod_annotation_prometheus_io_port`